modules/imgproc/src/grabcut.cpp

```diff
@@ -104,12 +111,51 @@ GMM::GMM( Mat& _model )
     mean = coefs + componentsCount;
     cov = mean + 3*componentsCount;
 
+    /* init memory for:
+     * double inverseCovs[componentsCount][3][3];
+     * double covDeterms[componentsCount];
+     * double sums[componentsCount][3];
+     * double prods[componentsCount][3][3];
+     * int sampleCounts[componentsCount];
+     */
+    inverseCovs = new double**[componentsCount];
+    prods = new double**[componentsCount];
+    sums = new double*[componentsCount];
+    sampleCounts = new int[componentsCount];
+    covDeterms = new double[componentsCount];
+    for( int ci = 0; ci < componentsCount; ci++ ) {
+        inverseCovs[ci] = new double*[3];
+        prods[ci] = new double*[3];
+        sums[ci] = new double[3];
+        for( int j = 0; j < 3; ++j ) {
+            inverseCovs[ci][j] = new double[3];
+            prods[ci][j] = new double[3];
+        }
+    }
```

Comment:
It generates too many new calls with small fragments. I propose to use std::vector<cv::Matx33d>. All Matx objects are allocated "in place". We'll get single allocation and simple indexing. You have to replace [] with () in several cases to follow Matx operators semantic.